File: engineering/codebase/smartqr.frontend-services/src/domain/codes/rules/enums/CodeRuleType.ts
/** Defines the role a rule plays in a code's routing — the wire discriminator for the rule hierarchy. */
export const CodeRuleType = {
  /** A rule matched against a scan signal, in order. */
  Conditional: "conditional",

  /** The catch-all serving its own content when no conditional rule matches. */
  Default: "default",

  /** The catch-all delegating to another rule's content. */
  DefaultPointer: "defaultPointer",
} as const;

export type CodeRuleType = (typeof CodeRuleType)[keyof typeof CodeRuleType];

File: engineering/codebase/smartqr.backend-services/SmartQr.Domain/Codes/Rules/Enums/CodeRuleType.cs
namespace SmartQr.Domain.Codes.Rules.Enums;

/// <summary>Defines the role a rule plays in a code's routing.</summary>
public enum CodeRuleType
{
    /// <summary>Represents a rule matched against a scan signal, in order.</summary>
    Conditional,

    /// <summary>Represents the catch-all serving its own content when no conditional rule matches.</summary>
    Default,

    /// <summary>Represents the catch-all delegating to another rule's content.</summary>
    DefaultPointer,
}
